packages/agent/src/agent/api.ts

```diff
@@ -161,11 +172,16 @@ export interface Agent {
    * @param canisterId The Principal of the Canister to send the query to. Sending a query to
    *     the management canister is not supported (as it has no meaning from an agent).
    * @param options Options to use to create and send the query.
+   * @param identity Sender principal to use when sending the query.
    * @returns The response from the replica. The Promise will only reject when the communication
    *     failed. If the query itself failed but no protocol errors happened, the response will
    *     be of type QueryResponseRejected.
    */
-  query(canisterId: Principal | string, options: QueryFields): Promise<QueryResponse>;
+  query(
+    canisterId: Principal | string,
+    options: QueryFields,
```

Comment:
do you think it makes sense to add `identity` to `options` instead of changing the function signature?

Reply:
I thought the same things when i was looking at this code, but i haven't modified anything, this was just added to the interface for completeness because the api query implementation was already receiving the identity as an extra argument [here](https://github.com/keplervital/agent-js/blob/8254c47694c1286058a8293547bda98a6d19c8e1/packages/agent/src/agent/http/index.ts#L377).